### PR TITLE
Acrs 15 save and exit

### PR DESCRIPTION
**Pull request by TemitopeAyokuHO**
## What? 
[ACRS-15](https://collaboration.homeoffice.gov.uk/jira/browse/ACRS-15) Save and Exit (returning user feature)

## Why? 
The save and exit functionality exists on several pages throughout the form. it helps user and save the current form and return in future to continue 

## How? 
- Added save and exit behaviour functionality to allow user to Save and Exit form when user decides to continue later in 
save-and-exit.js file
- Merged ACRS- 61 and ACRS-35 branch to display and test if they work ok 
- Added html pages to inform user the details has been saved

## Testing?
Tested on my local machine 

## Check list

- [x] I have reviewed my own pull request for linting issues (e.g. adding new lines)
- [ ] I have written tests (if relevant)
- [x] I have created a JIRA number for my branch
- [x] I have created a JIRA number for my commit
- [x] I have followed the chris beams method for my commit https://cbea.ms/git-commit/
here is an [example commit](https://github.com/UKHomeOfficeForms/hof/commit/810959f391187c7c4af6db262bcd143b50093a6e)
- [x] Ensure drone builds are green especially tests
- [x] I will squash the commits before merging

### CODE REVIEW

**Review by jamiecarterHO — CHANGES_REQUESTED** [review]
Just a couple of minor things to update copied from IMA.

Also I think you should add a step in acrs/index.js for information-saved with a save-and-exit behaviour like in IMA: https://github.com/UKHomeOffice/ima/blob/master/apps/ima/behaviours/save-and-exit.js. I know we don't send an email on save and exit but we should do the clear session bit.

File: apps/acrs/translations/src/en/validation.json [hunk 1]
```diff
@@ -31,5 +34,8 @@
     },
     "/how-send-decision": {
       "required": "You must select an option"
+    },
+    "id-type": {
+      "required": "You must select an option"
```
[the remark this answers is not in the document]
Comment (TemitopeAyokuHO) [reply apps/acrs/translations/src/en/validation.json:39]: done

File: apps/acrs/sections/summary-data-sections.js [hunk 2]
```diff
@@ -1,7 +1,82 @@
...
+        field: 'legal-representative-email',
+        parse: (list, req) => {
+          if (!req.sessionModel.get('steps').includes('/immigration-adviser-details')) {
+            return null;
+          }
+          return req.sessionModel.get('is-legal-representative-email') === 'yes' ?
+            `${req.sessionModel.get('user-email')}` :
+            `${req.sessionModel.get('legal-representative-email')}`;
+        },
+        omitChangeLink: true
```
[the remark this answers is not in the document]
Comment (TemitopeAyokuHO) [reply apps/acrs/sections/summary-data-sections.js:58]: Yes i reckon

File: apps/acrs/behaviours/continue-report.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      if (orderedSessionSteps.length < DEFAULTS.steps.length) {
+        req.sessionModel.set('steps', DEFAULTS.steps);
+        lastestStepInJourney = DEFAULTS.steps[DEFAULTS.steps.length - 1];
+      }
+
+      req.sessionModel.set('save-return-next-step', lastestStepInJourney);
+
+      return super.getValues(req, res, next);
+    }
+    return res.redirect('/ima/cases');
```
[the remark this answers is not in the document]
Comment (TemitopeAyokuHO) [reply apps/acrs/behaviours/continue-report.js:54]: done

**Comment by TemitopeAyokuHO** [comment]
> Just a couple of minor things to update copied from IMA.
> 
> Also I think you should add a step in acrs/index.js for information-saved with a save-and-exit behaviour like in IMA: https://github.com/UKHomeOffice/ima/blob/master/apps/ima/behaviours/save-and-exit.js. I know we don't send an email on save and exit but we should do the clear session bit.

I think these 2 steps were added already but should be in other ticket

File: apps/acrs/sections/summary-data-sections.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jamiecarterHO) [reply apps/acrs/sections/summary-data-sections.js:58]: Can update this in the other PR and I'll approve this one

**Review by jamiecarterHO — APPROVED** [review]
Approved assuming the save and exit step and behaviour are added in another PR as suggested.

**Pull request merged**